src/chunks/norms.rs

```diff
@@ -24,11 +18,29 @@ pub trait Norms {
 /// The unnormalized embedding can be reconstructed by multiplying the
 /// normalized embedding by its orginal l2 norm.
 #[derive(Clone, Debug)]
-pub struct NdNorms(pub Array1<f32>);
+pub struct NdNorms {
+    norms: Array1<f32>,
+}
+
+impl NdNorms {
+    /// Construct new `NdNorms`.
+    pub fn new<T>(norms: T) -> Self
```

Comment:
For these simple cases, I prefer:

```rust
pub fn new<T>(norms: impl Into<Array1<f32>>) -> Self
```

Since it's a trivial constructor, maybe it's also worthwhile to implement `From`?